podman: init at 1.0.0

vdemeester opened the pull request:
Signed-off-by: Vincent Demeester <vincent@sbr.pm>

###### Motivation for this change

podman is a binary build from libpod : libpod is a library used to
create container pods. podman aims to be *almost* compatible with the
docker cli but doesn't require a docker daemon.

If it uses a recent `runc` (aka 1.0.0-rc6), it support user namespaces containers (aka `podman run busybox` as a normal user would work :wink:)

See https://podman.io

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

nlewo commented on pkgs/applications/virtualization/podman/default.nix:
```diff
@@ -0,0 +1,55 @@
+    dontStrip = true;
+
+    buildPhase = ''
+    patchShebangs .
```
Indentation mistake (2 more spaces at the beginning of this line)

nlewo commented on pkgs/applications/virtualization/podman/default.nix:
```diff
@@ -0,0 +1,55 @@
+      btrfs-progs libseccomp gcc gpgme lvm2
+      ];
+
+    dontStrip = true;
```
Why?

nlewo commented on pkgs/applications/virtualization/podman/default.nix:
```diff
@@ -0,0 +1,55 @@
+    '';
+
+    installPhase = ''
+    install -Dm755 bin/podman $out/bin/podman
```
Indentation

nlewo commented on pkgs/applications/virtualization/podman/default.nix:
```diff
@@ -0,0 +1,55 @@
+
+  meta = {
+    homepage = https://podman.io/;
+    description = "libpod is a library used to create container pods.";
```
No dot at the end of the descritpion.
Maybe `A program for managing pods, containers, and container images` would be more better.

nlewo reviewed (changes requested):
Thanks a lot for bringing this into nixpkgs :) I definitely have to play with it!
Just some minor reviews.

bhipple commented on pkgs/applications/virtualization/podman/conmon.nix:
```diff
@@ -0,0 +1,33 @@
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [
+    makeWrapper libtool gcc glib
+  ];
```
Are you sure you need to specify all of these manually?  Normally they come in via `stdenv` by default.

nlewo replied:
No, they don't.

bhipple commented on pkgs/applications/virtualization/podman/default.nix:
```diff
@@ -0,0 +1,55 @@
+    install -Dm755 bin/podman $out/bin/podman
+    '';
+
+    outputs = ["out"];
```
If you just have a single output package I believe you can delete this.

bhipple commented:
Awesome!  I was thinking the other day about packaging this myself; glad to see others are interested in it too.

vdemeester commented:
@nlewo @bhipple updated :angel:

nlewo commented on pkgs/applications/virtualization/podman/conmon.nix:
```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchFromGitHub, makeWrapper, pkgconfig, libtool, gcc, glib }:
```
```suggestion
{ stdenv, lib, fetchFromGitHub, makeWrapper,  glib }:
```

nlewo commented:
Also, it would have been better to have two commits, one per package (it doesn't seem to me `conom` is a dep of `podman`.).

vdemeester committed:
conmon: init at unstable-2018-11-28

conmon is a required dependency for `podman`

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
@nlewo make sense :+1: updated :wink:

veprbl commented on pkgs/applications/virtualization/podman/default.nix:
```diff
@@ -0,0 +1,50 @@
+  
+  preFixup = ''
+    find $out -type f -exec remove-references-to -t ${go} -t ${stdenv.cc.cc} '{}' +
+    find $out -type f -exec remove-references-to -t ${stdenv.glibc.dev} '{}' +
```
Can't you combine this in a single command? You can put ` \` at the end of line to break lines in bash if they are too long.

veprbl commented on pkgs/applications/virtualization/podman/default.nix:
```diff
@@ -0,0 +1,50 @@
+  nativeBuildInputs = [ pkgconfig ];
+  
+  buildInputs = [
+    removeReferencesTo go btrfs-progs libseccomp gpgme lvm2
```
`removeReferencesTo` goes to `nativeBuildInputs`

veprbl commented on pkgs/applications/virtualization/podman/conmon.nix:
```diff
@@ -0,0 +1,33 @@
+  ];
+
+  installPhase = ''
+    install -D -m 755 bin/conmon $out/bin/conmon
```
We are writing to the store so we can set `-m555` from the beginning.

vdemeester committed:
podman: init at 0.11.1.1

podman is a binary build from libpod : libpod is a library used to
create container pods. podman aims to be *almost* compatible with the
docker cli but doesn't require a docker daemon.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester committed:
podman: add patch from containers/libpod#1977

Removes the `-i` from the `go build` commands. Once the PR is merged
and released, this patch won't be required anymore.

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
Updated with the rebase and patch from upstream (PR containers/libpod#1977) to fix build failure.
/cc @Mic92

vdemeester committed:
podman: 0.11.1 -> 0.12.1

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

vdemeester commented:
And bumped to `0.12.1` :angel:

vdemeester committed:
podman: 0.12.1 -> 0.12.1.2

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

Mic92 commented on pkgs/applications/virtualization/podman/conmon.nix:
```diff
@@ -0,0 +1,33 @@
+
+with lib;
+
+stdenv.mkDerivation rec {
```
Please use here `buildGoPackage` otherwise the result will depend on the go compiler.
You can still overwrite the build phase if needed.

Mic92 committed:
podman: use buildGoPackage

Mic92 commented:
Please always use `buildGoPackage` as it take care of removing references and works with cross-compiling.

Mic92 commented:
Please have a look if I missed something.

vdemeester commented:
@Mic92 yeah looks even better, thank you :heart_eyes_cat: 
LGTM ! :tada:

vdemeester commented:
ping @nlewo @Mic92

vdemeester committed:
podman: 0.12.1.2 -> 1.0.0

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

nlewo commented:
@GrahamcOfBorg build conmon podman

nlewo commented:
Thanks!
